diff -crB ./emma-node/platform/avr-raven/contiki-conf.h ./emma-node-new/platform/avr-raven/contiki-conf.h
*** ./emma-node/platform/avr-raven/contiki-conf.h	2013-09-03 17:52:44.793682590 +0200
--- ./emma-node-new/platform/avr-raven/contiki-conf.h	2013-09-03 17:53:32.185684106 +0200
***************
*** 333,339 ****
  #define UIP_CONF_ND6_REACHABLE_TIME     600000
  #define UIP_CONF_ND6_RETRANS_TIMER      10000
  /* For slow slip connections, to prevent buffer overruns */
! //#define UIP_CONF_RECEIVE_WINDOW 300
  #undef UIP_CONF_FWCACHE_SIZE
  #define UIP_CONF_FWCACHE_SIZE    30
  #define UIP_CONF_BROADCAST       1
--- 333,339 ----
  #define UIP_CONF_ND6_REACHABLE_TIME     600000
  #define UIP_CONF_ND6_RETRANS_TIMER      10000
  /* For slow slip connections, to prevent buffer overruns */
! #define UIP_CONF_RECEIVE_WINDOW 300
  #undef UIP_CONF_FWCACHE_SIZE
  #define UIP_CONF_FWCACHE_SIZE    30
  #define UIP_CONF_BROADCAST       1
